source/deployment/elasticsearch.rst

```diff
@@ -0,0 +1,78 @@
+Elasticsearch Beta (E20)
+========================
+
+*Available in Enterprise Edition E20.*
+
+`Elasticsearch <https://www.elastic.co/guide/en/elasticsearch/reference/current/index.html>`_ is a distributed, RESTful search engine capable of supporting search in large scale enterprise deployments with millions of messages. 
+
+.. toctree::
+    :maxdepth: 2
+
+Deployment Guide
+----------------
+
+Overview
+~~~~~~~~
+
+Elasticsearch allows you to search large volumes of data quickly, and in near real time, by creating and managing an index of post data. The indexing process can be managed from the System Console after setting up and connecting an Elasticsearch server. The post index is stored on the Elasticsearch server and is updated constantly after new posts are made. In order to index existing posts, a bulk index of the enitre post database must be generated. 
+
+Setting up an Elasticsearch Server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The setup process for the Elasticsearch server is documented in the `official Elasticsearch documentation <https://www.elastic.co/guide/en/elasticsearch/reference/current/setup.html>`_. 
+
+XXXXXX George: Any special steps specific to Mattermost implementation not mentioned in the offical docs? 
```

Comment:
Actually one thing. You need to install the ICU analyzer plugin. https://www.elastic.co/guide/en/elasticsearch/plugins/current/analysis-icu.html